pkg/lorry/plugin/proto/db_plugin.proto

```diff
@@ -0,0 +1,256 @@
+}
+
+message GetPluginInfoResponse {
+  // The name of the plugin. It MUST be 63
+  // characters or less, beginning and ending with an alphanumeric
+  // character ([a-z0-9A-Z]) with dashes (-), dots (.), and
+  // alphanumerics between. This field is REQUIRED.
+  string name = 1;
+
+  string db_type = 2;
+
+  // This field is REQUIRED. 
+  string version = 3;
+
+  // This field is OPTIONAL. 
+  map<string, string> metadata = 4;
+}
+
+message IsDBReadyRequest {
+  map<string, string> metadata = 1;
+}
+
+message IsDBReadyResponse {
+  // true if the DB is ready, false otherwise.
+  bool ready = 1;
+
+  // This field is OPTIONAL. 
+  map<string, string> metadata = 2;
+}
+
+message DBInfo {
+  // The pod FQDN of the replica.
+  string fqdn = 1;
+
+  // The port on which the DB service listens.
+  string port = 2;
+
+  // The admin username used to access the DB service.
+  string admin_user = 3;
+
+  // The admin password of the user used to access the DB service.
+  string admin_password = 4 [(kb_secret) = true];
+}
+
+message GetRoleRequest {
+  DBInfo db_info = 1;
+
+  // Common metadata property for extention
+  map<string, string> metadata = 2;
+}
+
+message GetRoleResponse {
+  string role = 1;
+
+  // Common metadata property for extention
+  map<string, string> metadata = 2;
+}
+
+message JoinMemberRequest {
+  DBInfo db_info = 1;
+
+  // The pod FQDN of the primary replica to check the role.
+  string primary_fqdn = 2;
+
+  // The port on which the primary DB service listens.
+  string primary_port = 3;
+
+  // The Names of all members.
+  repeated string members = 4;
+
+  // The name of the new member's Pod.
+  string new_member = 5;
+
+  // The IP of the new member's Pod.
+  string new_member_ip = 6;
+
+  // Common metadata property for extention
+  map<string, string> metadata = 7;
+}
+
+message JoinMemberResponse {
+  // Common metadata property for extention
+  map<string, string> metadata = 1;
+}
+
+message LeaveMemberRequest {
+  DBInfo db_info = 1;
+
+  // The pod FQDN of the primary replica to check the role.
+  string primary_fqdn = 2;
+
+  // The port on which the primary DB service listens.
+  string primary_port = 3;
+
+  // The Names of all members.
+  repeated string members = 4;
+
+  // The name of the leave member's Pod.
+  string leave_member = 5;
+
+  // The IP of the leave member's Pod.
+  string leave_member_ip = 6;
+
+  // Common metadata property for extention
+  map<string, string> metadata = 7;
+}
+
+message LeaveMemberResponse {
+  // Common metadata property for extention
+  map<string, string> metadata = 1;
+}
+
+message ReadonlyRequest {
+  DBInfo db_info = 1;
+
+  // The reason for setting the replica as read-only.
+  string Reason = 2;
+
+  // Common metadata property for extention
+  map<string, string> metadata = 3;
+}
+
+message ReadonlyResponse {
+  // Common metadata property for extention
+  map<string, string> metadata = 1;
+}
+
+message ReadwriteRequest {
+  DBInfo db_info = 1;
+
```

Comment:
To maintain symmetry with the Reason field in the ReadonlyRequest, there should be a corresponding field in this struct as well.